Objects/dictobject.c

```diff
@@ -1195,6 +1195,51 @@ dictresize(PyDictObject *mp, Py_ssize_t minsize)
     return 0;
 }
 
+/*
+ * clear dummy keys and reusing keys
+ */
+static void
+clear_dummy_keys(PyDictObject *mp)
+{
+    PyDictKeyEntry *first, *second, *end;
+    Py_ssize_t numentries;
+    PyDictKeyEntry *entries;
+
+    /* No DUMMY */
+    if (mp->ma_keys->dk_nentries == mp->ma_used) {
+        return;
+    }
+
+    numentries = mp->ma_used;
+    entries = DK_ENTRIES(mp->ma_keys);
+
+    first = entries;
+    second = entries + 1;
+    end = entries + numentries;
+
+    for( ;; ) {
```

Comment:
Use `for (;;) {` or `while (1) {`.
We always insert space outer parens, not inner.